sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/TransactionCoordinator.java

```diff
@@ -14,19 +15,23 @@
 import org.apache.qpid.proton.amqp.transport.DeliveryState;
 import org.apache.qpid.proton.engine.impl.DeliveryImpl;
 import org.apache.qpid.proton.message.Message;
+import reactor.core.Disposable;
 import reactor.core.publisher.Mono;
 
+import java.util.concurrent.atomic.AtomicBoolean;
+
 import static com.azure.core.amqp.implementation.ClientConstants.MAX_AMQP_HEADER_SIZE_BYTES;
 
 /**
  * Encapsulates transaction functions.
  */
-final class TransactionCoordinator {
+final class TransactionCoordinator implements AmqpTransactionCoordinator, Disposable {
```

Comment:
I would not use Disposable here. The annoying thing about this is that sender.dispose() is actually an async operation. So we should use `Mono<Void> dispose()`